document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/SubQueryFromExpression.java

```diff
@@ -0,0 +1,26 @@
+package org.hypertrace.core.documentstore.expression.impl;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Value;
+import org.hypertrace.core.documentstore.expression.type.FromTypeExpression;
+import org.hypertrace.core.documentstore.parser.FromTypeExpressionVisitor;
+import org.hypertrace.core.documentstore.query.Query;
+
+/**
+ * Allows embedding an entire Query in the FROM clause.
+ */
+@Value
+@Builder(toBuilder = true)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class SubQueryFromExpression implements FromTypeExpression {
```

Comment:
We would support either this or `JoinExpression`, right?

Reply:
Typically, this will be the `right` field in `JoinExpression`, as shown in the example in `JoinExpressionTest`